src/System.Net.Requests/tests/HttpWebRequestTest.cs

```diff
@@ -1098,18 +1106,24 @@ public void BeginGetRequestStream_CreatePostRequestThenCallTwice_ThrowsInvalidOp
             });
         }
 
-        [Theory, MemberData(nameof(EchoServers))]
-        public void BeginGetRequestStream_CreateRequestThenBeginGetResponsePrior_ThrowsProtocolViolationException(Uri remoteServer)
+        [Fact]
+        public async Task BeginGetRequestStream_CreateRequestThenBeginGetResponsePrior_ThrowsProtocolViolationException()
         {
-            HttpWebRequest request = HttpWebRequest.CreateHttp(remoteServer);
-
-            IAsyncResult asyncResult = request.BeginGetResponse(null, null);
-            Assert.Throws<ProtocolViolationException>(() =>
+            await LoopbackServer.CreateServerAsync((server, url) =>
             {
-                request.BeginGetRequestStream(null, null);
+                HttpWebRequest request = HttpWebRequest.CreateHttp(url);
+
+                IAsyncResult asyncResult = request.BeginGetResponse(null, null);
+                Assert.Throws<ProtocolViolationException>(() =>
+                {
+                    request.BeginGetRequestStream(null, null);
+                });
+
```

Comment:
Should we call request.EndGetResponse on the asyncResult and then dispose of it (after the Assert.Throws)?

Reply:
Calling EndGetResponse makes the test hang infinitely as it waits for a response which is never sent.

Reply:
That's because you've written the test to not send back a response.  You would need to fix that, too.